src/form-elements-regex.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * The regex used for matching `{ELEMENT:<elementName>}` tags in the OneBlink
+ * platform. Ignores nested elements based on the '|' character. Will match
+ * `"{ELEMENT:Parent_Name}"` but will NOT match `"{ELEMENT:Children|Name}"`.
+ */
+export const RootElementRegex = /({ELEMENT:)([^|}]+)(})/g
+/**
+ * The regex used for matching `{ELEMENT:<elementName>|<nestedElementName>}`
+ * tags in the OneBlink platform. Includes nested elements based on the '|'
+ * character. {ELEMENT:Children|Name} would be matched. Will match
+ * `"{ELEMENT:Parent_Name}"` and `"{ELEMENT:Children|Name}"`.
+ */
+export const NestedElementRegex = /({ELEMENT:)([^}]+)(})/g
+
+/**
+ * Takes a string and calls a provided handler function for each found instance
+ * of `{ELEMENT:<elementName>}` in the string. Used to replace values in
+ * OneBlink calculation and info (HTML) elements.
+ *
+ * #### Example
+ *
+ * ```js
+ * formElementsService.matchElementsTagRegex(
+ *   myString,
+ *   ({ elementName, elementMatch }) => {
+ *     const v = submission[elementName]
+ *     myString = myString.replace(elementMatch, v)
+ *   },
+ * )
+ * ```
+ *
+ * @param data
+ * @param handler
+ * @returns
+ */
+export function matchElementsTagRegex(
+  data: string,
+  matchHandler: (options: {
+    elementName: string
+    elementMatch: string
+  }) => void,
+  /**
+   * Determine if only root level elements should be matched.
+   *
+   * `false` will match `"{ELEMENT:Parent_Name}"` and
+   * `"{ELEMENT:Children|Name}"`.
+   *
+   * `true` will match `"{ELEMENT:Parent_Name}"` but will NOT replace
+   * `{ELEMENT:Children|Name}`.
+   */
+  rootFormElementsOnly: boolean,
```

Comment:
Obligatory comment about making this `includeNestedElements?: boolean` so getting the existing behaviour is not a breaking change.

Reply:
I don't like it but I am on board. Changing now.

Reply:
However, to achieve the existing functionality, the variable would need to be `excludeNestedElements`.